src/python/pants/backend/openapi/dependency_inference.py

```diff
@@ -0,0 +1,175 @@
+
+                if not normalized.startswith("../"):
+                    local_refs.add(normalized)
+
+    return frozenset(local_refs)
+
+
+# -----------------------------------------------------------------------------------------------
+# `openapi_definition` dependency inference
+# -----------------------------------------------------------------------------------------------
+
+
+@dataclass(frozen=True)
+class OpenApiDefinitionDependenciesInferenceFieldSet(FieldSet):
+    required_fields = (OpenApiDefinitionField,)
+
+    sources: OpenApiDefinitionField
+
+
+class InferOpenApiDefinitionDependenciesRequest(InferDependenciesRequest):
+    infer_from = OpenApiDefinitionDependenciesInferenceFieldSet
+
+
+@rule
+async def infer_openapi_definition_dependencies(
+    request: InferOpenApiDefinitionDependenciesRequest,
+) -> InferredDependencies:
+    hydrated_sources = await Get(HydratedSources, HydrateSourcesRequest(request.field_set.sources))
+    candidate_targets = await Get(
+        Targets,
+        RawSpecs(
+            file_literals=(FileLiteralSpec(*hydrated_sources.snapshot.files),),
+            description_of_origin="the `openapi_definition` dependency inference",
+        ),
+    )
+
+    addresses = [
+        target.address for target in candidate_targets if target.has_field(OpenApiSourceField)
+    ]
+
+    return InferredDependencies(addresses)
+
+
+# -----------------------------------------------------------------------------------------------
+# `openapi_source` dependency inference
+# -----------------------------------------------------------------------------------------------
+
+
+@dataclass(frozen=True)
+class OpenApiSourceDependenciesInferenceFieldSet(FieldSet):
+    required_fields = (OpenApiSourceField,)
+
+    sources: OpenApiSourceField
+
+
+class InferOpenApiSourceDependenciesRequest(InferDependenciesRequest):
+    infer_from = OpenApiSourceDependenciesInferenceFieldSet
+
+
+@rule
+async def infer_openapi_module_dependencies(
+    request: InferOpenApiSourceDependenciesRequest,
+) -> InferredDependencies:
+    hydrated_sources = await Get(HydratedSources, HydrateSourcesRequest(request.field_set.sources))
+    result = await Get(
+        OpenApiDependencies,
+        ParseOpenApiSources(
+            sources_digest=hydrated_sources.snapshot.digest,
+            paths=hydrated_sources.snapshot.files,
+        ),
+    )
+
+    paths: set[str] = set()
+
+    for source_file in hydrated_sources.snapshot.files:
+        paths.update(result.dependencies[source_file])
+
+    candidate_targets = await Get(
+        Targets,
+        RawSpecs(
+            file_literals=tuple(FileLiteralSpec(path) for path in paths),
+            unmatched_glob_behavior=GlobMatchErrorBehavior.ignore,
+            description_of_origin="the `openapi_source` dependency inference",
+        ),
+    )
+
+    addresses = [
+        target.address for target in candidate_targets if target.has_field(OpenApiSourceField)
+    ]
```

Comment:
you may want to query for the explicitly provided dependencies, as users can se exclude rules in the `dependencies` field and that should be used to disambiguate cases in which you may find more than one target providing with the same dependency.

Check the following link to see an example on how it can be used:
https://github.com/pantsbuild/pants/blob/bba429d42ed6521cb378ef0d069e3bfd1f4994fc/src/python/pants/backend/helm/dependency_inference/chart.py#L107

Reply:
I've added something similar, please have a look :)